extensions/ql-vscode/src/query-history.ts

```diff
@@ -191,12 +192,26 @@ export class QueryHistoryManager {
 
   async handleShowQueryLog(queryHistoryItem: CompletedQuery) {
     if (queryHistoryItem.logFileLocation) {
+      const uri = vscode.Uri.parse(queryHistoryItem.logFileLocation);
       try {
-        await vscode.window.showTextDocument(vscode.Uri.parse(queryHistoryItem.logFileLocation), {
-          viewColumn: vscode.ViewColumn.Beside
+        await vscode.window.showTextDocument(uri, {
         });
       } catch (e) {
-        helpers.showAndLogErrorMessage(`Could not open log file ${queryHistoryItem.logFileLocation}`);
+        if (e.message.includes('Files above 50MB cannot be synchronized with extensions')) {
```

Comment:
I was going to ask if there was any other signal the exception is the appropriate one to react to other than scraping its message, but it looks like `src/vs/workbench/api/browser/mainThreadDocuments.ts` is throwing an `Error`, not anything more specific :(

Reply:
Yeah, I was looking for something better, but couldn't find it. And the worst thing is that it looks like it's not _actually_ 50Mb. There are some other triggers for this (like length of token stream), even though the message is the same AFAICT.